spack-repo/packages/singularity-eos_deps/package.py

```diff
@@ -1,38 +1,123 @@
-# dependency package for singulary
+# dependency package for singulary-eos
 
 import os
 from spack import *
 
-class SingularityEosDeps(BundlePackage):
+class SingularityEosDeps(BundlePackage, CudaPackage):
     homepage    = "https://github.com/lanl/singularity-eos"
     url         = "https://github.com/lanl/singularity-eos/archive/refs/heads/main.zip"
     git         = "git@github.com:lanl/singularity-eos.git"
 
     version("main", branch="main")
 
-    variant("use_cuda", default=False, description="Enable cuda support")
+    variant("kokkos",
+            description="Enable kokkos",
+            default=True
+    )
 
-    depends_on("cmake", type="build")
-    depends_on("hdf5~mpi+cxx+hl", type=("build", "run"))
-    depends_on("eigen@3.3.9", type="build")
-    depends_on("kokkos@3:", when="+use_cuda", type=("build", "run"))
-    depends_on("kokkos-kernels", when="+use_cuda", type=("build", "run"))
-    depends_on("eospac", type=("build","run"))
+    variant("kokkos-kernels",
+            description="Enable kokkos-kernals for linear algebra",
+            default=False
+    )
+
+    variant("openmp",
+            description="Enable openmp",
+            default=False
+    )
+
+    variant("build_extra",
+            description="Build converters",
+            values=any_combination_of(
+                'sesame','stellarcollapse'
+            ).with_default('none') 	
+    )
+
+    variant("enable_tests",
+            default=False,
+            description="Build tests"
+    )
+
+    variant("enable_fortran",
+            default=True,
+            description="Enable building fortran interface"
+    )
+
+    depends_on("mpark-variant")
+    depends_on("hdf5~mpi+cxx+hl")
```

Comment:
This might actually be a bug... downstream, I expect the host codes will want MPI turned on. That said, for stand-alone use, shouldn't matter.

Reply:
This is an artifact that's present to satisfy a peculiarity in the CMake configuration, and to enable tests that rely on it. Or, more simply, it's there because I wanted to cover the case of using HDF5 without the rigamarole of MPI. I'll take it out.